### PR TITLE
feat: Product-Support reference block

### LINKED ISSUE #93: Implement Product Support reference block
In most (if not all) product pages, there is a block (see e.g., https://www.mammotome.com/products/mammotome-revolve/) that references the image of the product and offers 3 links for the product ("product information", "product support", "product documents"). 

Bildschirm­foto 2023-05-02 um 15 37 28

We should provide such a block that auto-populates from the product db (see #67).

### PR DESCRIPTION
**Pull request by karlpauls**
Fix #93 

Test URLs:
- Before: https://main--mammotome--hlxsites.hlx.page/en/products/mammotome-confirm
- After: https://93-product-ref--mammotome--hlxsites.hlx.page/en/products/mammotome-confirm

### CODE REVIEW

File: blocks/product-reference/product-reference.js [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+    const translation = json.ProductTranslation.data
+      .find((entry) => entry.ProductRef === product.ProductCodes && entry.Language === language);
+
+    product.Name = translation?.Name || product.Name;
+    product.Image = translation?.Image || product.Image;
+  }
+
+  return product;
+}
+
```
Comment (solaris007) [inline blocks/product-reference/product-reference.js:23]: for modularity / readability, you could add here a `createButtons` function:

```
async function createButtons() {
  const buttonData = [
    ['productReferenceInformationURL', '../contact/', 'productReferenceInformation', 'Request Information', 'primary'],
    ['productReferenceSupportURL', '../product-support', 'productReferenceSupport', 'Product Support', 'secondary'],
    ['productReferenceSupportURL', '../product-support', 'productReferenceDocuments', 'Product Documents', 'secondary']
  ];

  return Promise.all(buttonData.map(async ([urlKey, defaultUrl, textKey, defaultText, className]) => {
    const href = await translate(urlKey, defaultUrl);
    const textContent = await translate(textKey, defaultText);
    return { href, textContent, className };
  }));
}
```

File: blocks/product-reference/product-reference.js [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+  const { language } = getInfo();
+  const json = await getProductDB();
+  const product = getProduct(json, productCode, language);
+
+  if (!product) {
+    return;
+  }
+
+  const buttons = [
```
Comment (solaris007) [inline blocks/product-reference/product-reference.js:41]: then you can shorten the following with this:

```
  const buttons = await createButtons();
  const buttonStructure = buttons.map(({ href, textContent, className }) => ({
    type: 'a',
    attributes: { href },
    children: [{ type: 'button', classes: [className], textContent }]
  }));

  createDomStructure([
    {
      type: 'div',
      children: [
        {
          type: 'div',
          classes: ['product-ref-container'],
          children: [{ type: 'img', attributes: { src: product.Image } }]
        },
        ...buttonStructure.map(child => ({
          type: 'div',
          classes: ['product-ref-container'],
          children: [{ type: 'div', children: [child] }]
        }))
      ]
    }
  ], block);
```

**Comment by karlpauls** [comment]
@solaris007, I see where you are going with this - it was missing some subtleties but I tried to follow the gist of it. Certainly worthwhile just to hopefully make it more clear what is going on (i.e., the product code is needed for the button data).

**Review by solaris007 — APPROVED** [review]
LGTM

**Pull request merged**